model setup

ritachien opened the pull request:
**U37 進度: 所有 models 基本欄位、關聯設置完成**
* 通過欄位檢查測試
* 通過關聯性檢查測試
* 通過 CRUD 檢查測試
* Users 新增 account、cover 欄位(account-註冊及登入、cover-前端畫面)

**其他:**
* 將 eslint 載入專案。commit 部分應該不用特別看，除了 model/index 改了一個 `path.resolve()` 之外，其他都是 format 撰寫格式而已。 
* 我覺得之前 commit 太多太亂，反正你還沒 code review，所以我直接刪掉分支重新發 PR。這樣你讀 commit 內容才不會太亂!

ritachien committed:
feat: modify model initial settings

ritachien committed:
feat: add account/cover columns to Users

ritachien committed:
feat: add eslint to project

ritachien committed:
feat: add passport-jwt login feature

ritachien committed:
fix: add account/cover to User model

ritachien commented:
**登入驗證功能完成:**
* 使用 passport-jwt 進行登入驗證，目前只有做簡單的資料檢查。之後開發過程應該陸續還要加上更多資料驗證。

**其他:**
* 之前 migration 有增加 User 的兩個欄位，但 model 部分沒有加到，這次 push 已修正。

ritachien commented:
![](https://github.trello.services/images/mini-trello-icon.png) [後端專案基礎設置](https://trello.com/c/Olh9KM2n/11-%E5%BE%8C%E7%AB%AF%E5%B0%88%E6%A1%88%E5%9F%BA%E7%A4%8E%E8%A8%AD%E7%BD%AE)

rubylo718 reviewed (commented):
~~Hi Rita, please add "underscord: true" in your every model.~~
We do not add "underscored: true" to meet the spec in test files after confirmation.

rubylo718 commented on middleware/auth.js:
```diff
@@ -0,0 +1,19 @@
+  return res.json({
+    status: 'success',
+    message: 'permission denied'
+  })
```
success but permission denied?

ritachien replied:
Fixed!
```js
status: 'success'  => status: 'error'
```

rubylo718 reviewed (commented):
please check middleware/auth.js 
Line 11-12: status is success, but permission denied?

ritachien commented:
In middleware/auth.js
Should I add another function to explicitly check auth for general user?
```js
const authenticatedUser = (req, res, next) => {
  if (helpers.getUser(req)?.role === 'user') return next()
  return res.json({
    status: 'error',
    message: 'permission denied'
  })
}
```

ritachien committed:
fix: edit error status

rubylo718 commented:
yes. please go.

ritachien committed:
feat: separate user/admin signin route

user story 要求前台和後台登入頁面分離，不同身分在對方的登入頁面
無法成功登入。為了將路由功能區分得更明確並多一層檢查，所以將登入路
郵拆成 admin/user 兩條。

ritachien commented:
**修改內容:**
1. 前面提到的 error status
2. 登入路由拆成 admin/user 兩條(和前端討論結果)

ritachien committed:
fix: skip .trim() if variable not in req.body

ritachien commented:
It would pop error if trim target isn't in request body. Fixed this bug in `b50e98e` commit.

rubylo718 reviewed (commented):
1. user-controller signup: please align the variable name with FE (passwordCheck or checkPassword)
2. middleware/auth.js: the variable name isXXX would be intended to be a boolean. Please consider to change the names of middleware isAdmin and isUser.

ritachien committed:
fix: modify variables name, controller and route

variable name - for meet frontend naming
userController - fix user find range to avoid name doubled
route - edit signup route

ritachien commented:
**user-controller signup:**
1. align the variable name with FE
2. add trim() to check if input data blank
3. edit check range(from email to name) to avoid same value with other user

**middleware/auth.js:**
edit variable name( isXXX --> authXXX)

**users route:**
fix signup route

rubylo718 reviewed (approved):
Everything looks good good!
